src/main/java/seedu/duke/Storage.java

```diff
@@ -0,0 +1,194 @@
+     * @param date given by user input
+     * @return LocalDate object.
+     */
+    public static LocalDate getLocalDate(String date) {
+        String[] dateDetails = date.split("-");
+        String day = dateDetails[0];
+        String month = dateDetails[1];
+        String year = dateDetails[2];
+        if (day.length() == 1) {
+            day = "0" + day;
+        }
+        if (month.length() == 1) {
+            month = "0" + month;
+        }
+        List<String> list = Arrays.asList(year, month, day);
+
+        String dateToParse = String.join("-", list);
+        System.out.println(dateToParse);
+        return LocalDate.parse(dateToParse);
+    }
+
+    /**
+     * Adds ToDo object to tasklist in TaskList.
+     * @param taskList this.tasklist of TaskList object.
+     * @param isMarked true if the task being read is marked.
+     * @param description Task description.
+     */
+    public void addToDo(ArrayList<Task> taskList, String isMarked, String description) {
+//        System.out.println("added T");
+        ToDo todo = new ToDo(description);
+        if (isMarked.equals("1")) {
+            todo.mark();
+        }
+        taskList.add(todo);
+    }
+
+    /**
+     * Adds Deadline or Event object to tasklist in TaskList.
+     * @param taskList this.tasklist of TaskList object.
+     * @param s String differentiates between a Deadline or Event object.
+     * @param isMarked true if the task being read is marked.
+     * @param description Task description.
+     * @param datetime String datetime based on user input.
+     */
+    public void addDeadlineOrEvent(ArrayList<Task> taskList, String s, String isMarked, String description, String datetime){
+//        System.out.println("type:" + s);
+        if (s.equals("D")){
+//            System.out.println("added D");
+            String[] dateAndTime = datetime.split(" ");
+            System.out.println(dateAndTime.length);
+            if (dateAndTime.length == 2) {
+                String dueDate = dateAndTime[0];
+                String dueTime = dateAndTime[1];
+                LocalDate localDate = getLocalDate(dueDate);
+                Task deadline = new Deadline(description, localDate, dueTime);
+                if (isMarked.equals("1")) {
+                    deadline.mark();
+                }
+                taskList.add(deadline);
+            } else {
+                String dueDate = dateAndTime[0];
+                LocalDate localDate = getLocalDate(dueDate);
+                Task deadline = new Deadline(description, localDate);
+                if (isMarked.equals("1")) {
+                    deadline.mark();
+                }
+                taskList.add(deadline);
+            }
+        } else {
+//            System.out.println("added E");
+            Event event = new Event(description, datetime);
+            if (isMarked.equals("1")) {
+                event.mark();
+            }
+            taskList.add(event);
+        }

```

Comment:
Have you considered using switch cases as well here since the conditionals are a little deeply nested?